devito/passes/clusters/blocking.py

```diff
@@ -161,3 +203,82 @@ def decompose(ispace, d, block_dims):
     directions.update({bd: ispace.directions[d] for bd in block_dims})
 
     return IterationSpace(intervals, sub_iterators, directions)
+
+
+class Skewing(Queue):
+
+    """
+    Construct a new sequence of clusters with skewed expressions and iteration spaces.
+
+    Notes
+    -----
+    This transformation is applying loop skewing to derive the
+    wavefront method of execution of nested loops. Loop skewing is
+    a simple transformation of loop bounds and is combined with loop
+    interchanging to generate the wavefront [1]_.
+
+    .. [1] Wolfe, Michael. "Loops skewing: The wavefront method revisited."
+    International Journal of Parallel Programming 15.4 (1986): 279-293.
+
+    Examples:
+
+    .. code-block:: python
+
+        for i = 2, n-1
+            for j = 2, m-1
+                a[i,j] = (a[a-1,j] + a[i,j-1] + a[i+1,j] + a[i,j+1]) / 4
+
+    to
+
+    .. code-block:: python
+
+        for i = 2, n-1
+            for j = 2+i, m-1+i
+                a[i,j-i] = (a[a-1,j-i] + a[i,j-1-i] + a[i+1,j-i] + a[i,j+1-i]) / 4
+
+    """
+
+    def __init__(self, options):
+        self.skewinner = bool(options['blockinner'])
+
+        super(Skewing, self).__init__()
+
+    def callback(self, clusters, prefix):
+        if not prefix:
+            return clusters
+
+        d = prefix[-1].dim
+
+        processed = []
+        for c in clusters:
+            # Explore skewing possibilities only in case d is SKEWABLE
+            # and not innermost
+            if SKEWABLE not in c.properties[d]:
+                return clusters
+
+            if d is c.ispace[-1].dim and not self.skewinner:
+                return clusters
+
+            # Search for a skew_dim candidate, if more than one, we keep the last
```

Comment:
that's not what it's happening below, because if len()>1 you return. You may even drop  this comment

Reply:
right!